src/BitStringAddresses/bosefs.jl

```diff
@@ -230,29 +207,44 @@ end
 end
 
 """
-    occupied_orbitals(b)
-
-Iterate over occupied orbitals in `BoseFS` address. Returns tuples of
-`(boson_number, orbital_number, bit_offset)`.
+    BoseFSIndex
 
-Note that the `bit_offset` is zero-based!
+Convenience struct for indexing into a fock state.
 
-# Example
+## Fields:
 
-```jldoctest
-julia> b = BoseFS((1,5,0,4))
-julia> for (n, i) in occupied_orbitals(b)
-    @show n, i
+* `occnum`: the occupation number.
+* `site`: the index of the site.
+* `offset`: the bit offset of the site.
+"""
+struct BoseFSIndex <: FieldVector{3,Int}
+    occnum::Int
+    site::Int
+    offset::Int
 end
-(n, i) = (1, 1)
-(n, i) = (5, 2)
-(n, i) = (4, 4)
 
-"""
 struct OccupiedOrbitalIterator{C,S}
     address::S
 end
 
+"""
+    occupied_orbitals(b::BoseFS)
+
+Iterate over occupied orbitals in `BoseFS` address. Iterates values of type
+[`BoseFSIndex`](@ref).
```

Comment:
I'm not sure whether we'll ever need it, but to make this a proper iterator it would be good to have a method for `Base.length`.